### PR TITLE
PYTHON-3159 Fix typo in zlib compression support

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: pymongo/compression_support.py [hunk 1]
```diff
@@ -123,16 +122,10 @@ class ZlibContext(object):
...
-        self.compress: Callable[[bytes], bytes]
-
-        # Jython zlib.compress doesn't support -1
-        if level == -1:
-            self.compress = zlib.compress
-        # Jython zlib.compress also doesn't support 0
-        elif level == 0:
-            self.compress = _zlib_no_compress
-        else:
-            self.compresss = lambda data, _: zlib.compress(data, level)
```
Comment (ShaneHarvey) [inline pymongo/compression_support.py:135]: `compresss` -> `compress` was the bug fix. I also removed the Jython workarounds.

File: pymongo/compression_support.py [hunk 2]
```diff
@@ -123,16 +122,10 @@ class ZlibContext(object):
     compressor_id = 2
 
     def __init__(self, level):
-        self.compress: Callable[[bytes], bytes]
```
Comment (juliusgeo) [inline pymongo/compression_support.py:126]: Why as this code removed as opposed to moving it into `def compress` below? Is this logic no longer necessary for Jython compatibility?

Comment (ShaneHarvey) [reply pymongo/compression_support.py:126]: We don't support Jython.

Comment (juliusgeo) [reply pymongo/compression_support.py:126]: Ah, I see your comment below addresses part of this. I'm still curious why the Jython workarounds are no longer necessary, did we drop support for Jython?

**Review by juliusgeo — APPROVED** [review]
LGTM once tests pass.

**Pull request merged**